src/csharp/Grpc.Tools/ProtoCompile.cs

```diff
@@ -375,60 +375,48 @@ protected override bool ValidateParameters()
         static readonly Encoding s_utf8WithoutBom = new UTF8Encoding(false);
         protected override Encoding ResponseFileEncoding => s_utf8WithoutBom;
 
-        // Protoc takes one argument per line from the response file, and does not
-        // require any quoting whatsoever. Otherwise, this is similar to the
-        // standard CommandLineBuilder
-        class ProtocResponseFileBuilder
+        class ProtocCommandLineBuilder : CommandLineBuilder
         {
-            StringBuilder _data = new StringBuilder(1000);
-            public override string ToString() => _data.ToString();
-
-            // If 'value' is not empty, append '--name=value\n'.
-            public void AddSwitchMaybe(string name, string value)
+            // If 'value' is not empty, append '--name=value'.
+            public void AppendSwitchUnquotedIfNotNullOrEmpty(string name, string value)
             {
                 if (!string.IsNullOrEmpty(value))
                 {
-                    _data.Append("--").Append(name).Append("=")
-                         .Append(value).Append('\n');
+                    AppendSwitchIfNotNull($"--{name}=", value);
                 }
             }
 
             // Add switch with the 'values' separated by commas, for options.
-            public void AddSwitchMaybe(string name, string[] values)
+            public void AppendSwitchIfNotNullOrEmpty(string name, string[] values)
             {
                 if (values?.Length > 0)
                 {
-                    _data.Append("--").Append(name).Append("=")
-                         .Append(string.Join(",", values)).Append('\n');
+                    AppendSwitchIfNotNull($"--{name}=", values, ",");
                 }
             }
+        }
 
-            // Add a positional argument to the file data.
-            public void AddArg(string arg)
-            {
-                _data.Append(arg).Append('\n');
-            }
-        };
-
-        // Called by the base ToolTask to get response file contents.
-        protected override string GenerateResponseFileCommands()
+        // Called by the base ToolTask to get the command line.
+        protected override string GenerateCommandLineCommands()
         {
-            var cmd = new ProtocResponseFileBuilder();
-            cmd.AddSwitchMaybe(Generator + "_out", TrimEndSlash(OutputDir));
-            cmd.AddSwitchMaybe(Generator + "_opt", OutputOptions);
-            cmd.AddSwitchMaybe("plugin=protoc-gen-grpc", GrpcPluginExe);
-            cmd.AddSwitchMaybe("grpc_out", TrimEndSlash(GrpcOutputDir));
-            cmd.AddSwitchMaybe("grpc_opt", GrpcOutputOptions);
+            var cmd = new ProtocCommandLineBuilder();
```

Comment:
I'm worried about the switch from ResponseFile to CommandLine args. 
Can't this:
1. cause problems because we need to use quoting now?
2. cause issues like command line ending up too long etc.?

I'm actually not entirely sure why ResponseFile way of passing arguments was originally used.

Reply:
Usually, this is called with one proto file per call (even though Protobuf would allow for multiple items). So it would only get too long if too many proto paths are appended. So yes, the response file would be preferable. But it seems, that protoc won't get fixed soon, so this would only be good as workaround and should be reverted once protoc is fixed.

Reply:
I'm concerned about the second point here since we have had issues with long paths. Even in the case where there are a few ProtoPath specified, if each contains a long file path, the long path can cause issues. I think it's better to use response files and fix the issue in protoc instead. If we must accept this workaround, I would add an option to choose between response files and command line with a default to use response files.